### ISSUE TITLE
Build failure with mtl-2.3

### ISSUE THREAD

**Issue by sjakobi** [luntain/error-or-bundle#1]
```
src/Data/ErrorOr.hs:78:17: error:
    Ambiguous occurrence ‘mapError’
    It could refer to
       either ‘Control.Monad.Except.mapError’,
              imported from ‘Control.Monad.Except’ at src/Data/ErrorOr.hs:46:1-41
              (and originally defined in ‘Control.Monad.Error.Class’)
           or ‘Data.ErrorOr.mapError’,
              defined at src/Data/ErrorOr.hs:176:1
   |
78 |   | otherwise = mapError (ErrTag str) res
   |                 ^^^^^^^^
```

**Comment by sjakobi** [comment]
As a Hackage trustee I have revised error-or-0.2.0.0: https://hackage.haskell.org/package/error-or-0.2.0.0/revisions/

**Comment by luntain** [comment]
Thanks @sjakobi for changing the bounds to correct the issue. The `mtl-2.3` incompatibility should be fixed in ver 0.3.0, just uploaded to Hackage.
